client/configuration.py

```diff
@@ -396,6 +401,8 @@ def _read(self, path: str) -> None:
 
                 binary = configuration.consume("binary", current=self._binary)
                 assert binary is None or isinstance(binary, str)
+                if isinstance(binary, str):
```

Comment:
Nit: `binary is not None` reads a bit more cleanly to me :)

Reply:
Great point, that's a faster check too - thanks!